lms/djangoapps/courseware/views/views.py

```diff
@@ -513,6 +513,13 @@ def create_page_context(self, request, course=None, tab=None, **kwargs):
             request.user = masquerade_user
         else:
             masquerade = None
+
+        now = datetime.now(UTC())
+        effective_start = _adjust_start_date_for_beta_testers(request.user, course, course.id)
+        if not in_preview_mode() and staff_access and now < effective_start:
```

Comment:
I have a question about each of these cases:
1. `not in_preview_mode()`: Can you describe when someone would be in preview mode and would not want to set this to False?  It seems like someone could be previewing, and then make a change in Studio that renders this not visible, and then it might break because you were already in preview mode?  Not sure because I don't know what this is used for.
2. Can you explain why we need to check `staff_access` before determining whether or not to turn off preview?
3. `now < effective_start`: I've got sub-questions on this one:
- Seems like this logic should be central and probably already is used somewhere?
- Is it a problem to have this logic separate from has_access?  Could it be more integrated?
- Is there logic to limit the options in the preview menu to only types of learners that could have access, like cohorted users, or individual users with or without access?  What do we do when they don't have access?  Maybe this is a separate issue entirely, but I'm trying to minimize redundant logic.